event-service-rest/src/main/java/com/restapi/rest/MainController.java

```diff
@@ -1,13 +1,141 @@
+                .place(event.getPlace())
+                .speaker(event.getSpeaker())
+                .dateTime(event.getDateTime())
+                .build(),
+                HttpStatus.CREATED);
+    }
+
+    /**
+     * Get all events from the storage.
+     * @return
+     */
+    @GetMapping("/events")
+    public ResponseEntity<List<EventApi>> getEventList() {
+
+        List<EventApi> events = eventService.getAllEvents()
+                .stream().map(e -> new EventApi(e.getId(),
+                        e.getTitle(),
+                        e.getPlace(),
+                        e.getSpeaker(),
+                        e.getEventType(),
+                        e.getDateTime()))
+                .collect(Collectors.toList());
+
+        if (events.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+
+        return new ResponseEntity<>(events,
+                HttpStatus.OK);
+    }
+
+    /**
+     * Get one event by id
+     * @param id
+     * @return
+     */
+    @GetMapping("/events/{id}")
+    public ResponseEntity<EventApi> getById(@PathVariable Integer id) {
+
+        Event event = eventService.getEvent(id);
+
+        if (event == null) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+
+        return new ResponseEntity<>(EventApi.builder()
+                .id(event.getId())
+                .title(event.getTitle())
+                .eventType(event.getEventType())
+                .place(event.getPlace())
+                .speaker(event.getSpeaker())
+                .dateTime(event.getDateTime())
+                .build(),
+                HttpStatus.OK);
+    }
+
+    /**
+     * Update event by id
+     * @param id
+     * @return
+     */
+    @PutMapping("/events/{id}")
+    public ResponseEntity<EventApi> updateEvent(@PathVariable Integer id, @RequestBody EventApi payload) {
+
+        Event event = payload.getEvent();
+        event = eventService.updateEvent(event);
+
+
+        return new ResponseEntity<>(EventApi.builder()
+                .id(event.getId())
+                .title(event.getTitle())
+                .eventType(event.getEventType())
+                .place(event.getPlace())
+                .speaker(event.getSpeaker())
+                .dateTime(event.getDateTime())
+                .build(),
+                HttpStatus.CREATED);
     }
+
+    /**
+     * Delete event by id
+     * @param id
+     * @return
+     */
+    @DeleteMapping("/events/{id}")
+    public ResponseEntity<HttpStatus> deleteEvent(@PathVariable Integer id) {
+        Event event = eventService.getEvent(id);
+        if (event == null) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+
+        try {
+            eventService.deleteEvent(event);
+
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
are you sure you need to catch any exceptions here? will Spring not handle persistance failures?